pingora-core/src/protocols/http/v2/server.rs

```diff
@@ -387,13 +387,15 @@ impl HttpSession {
 
     /// Whether there is no more body to read
     pub fn is_body_done(&self) -> bool {
-        self.request_body_reader.is_end_stream()
+        // Check no body in request
+        // Also check we hit end of stream if request carry body.
+        self.is_body_empty() || (self.body_read > 0 && self.request_body_reader.is_end_stream())
     }
 
-    /// Whether there is any body to read.
+    /// Whether there is any body to read. false means there no body in request.
```

Comment:
`false means there no body in request`

We'll want to change this to `true means there's no body in request`.